### ISSUE TITLE
Publish pre-build bundles to npm

### ISSUE THREAD

**Issue by bershanskiy** [mdn/browser-compat-data#7374]
Please consider "bundling" all of the data for releases on npm to improve execution speed, reduce size of dependents' `node_modules` folder, and support legacy NodeJS versions.

Consider the following simple script that just loads BCD and stores all of it in a single file `export.json`.
```
'use strict';
const fs = require('fs');
const bcd = require('@mdn/browser-compat-data');
const string = JSON.stringify(bcd);
fs.writeFileSync('export.json', string);
```

Let's compare the current `@mdn/browser-compat-data` release and its "bundled" version. A minimal "bundled" version could include `export.json` produced by the script above and BCD's `LICENSE`, `README.md`, and `package.json`.

### Load time
Currently, `index.js` walks the directories and loads data files synchronously because it needs to be `require`'d by dependents. Numerous synchronous directory lookups and file reads cause it to be slow.

I flushed the file system cache before each measurement to make comparisons as fair as possible. I measured time during a few runs and results fluctuated a bit. On SSD, "real" time of `bundled.js` never exceeded 0.3s while "real" time of `current.js` never dropped below 1s. On HDD, "real" time of `bundled.js` never exceeded 1s while "real" time of `current.js` never dropped below 4s. (The "user" time of 0.000s looks suspicious to me, but it might be just Windows miss-reporting very small time intervals. The "real" time measurements do feel reasonable, especially the ones over 1s.)

#### Tests
```
$ cat current.js
'use strict';
const a = require('@mdn/browser-compat-data');

$ cat bundled.js
'use strict';
const a = require('./export.json');
```

#### Sample results
On a modern NVMe SSD:
```
$ time node bundled.js
real    0m0.265s
user    0m0.000s
sys     0m0.015s

$ time node current.js
real    0m1.089s
user    0m0.000s
sys     0m0.015s
```

On a slow HDD:
```
$ time node bundled.js
real    0m0.521s
user    0m0.000s
sys     0m0.000s

$ time node current.js
real    0m4.606s
user    0m0.015s
sys     0m0.030s
```

### File size
Size of `node_modules` containing current `@mdn/browser-compat-data` and `extend` (required to load BCD): 18,333,922 bytes
Size of minimal "bundled" version: 8,989,309 bytes

Size of minimal "bundled" version is the sum of size of `export.json` produced by the script above and sizes of BCD's `LICENSE`, `README.md`, and `package.json`.

The "bundle" comes out to be less than 50% of the original size!
Note: of course, `package.json` size would change slightly, but that change would be negligible.

### Legacy NodeJS versions
It's nice that BCD scripts and various linters use modern JS and modern version of NodeJS. Unfortunately, some dependents can't drop old NodeJS versions and are stuck with `mdn-browser-compat-data`.

Currently, BCD requires NodeJS 10 or above, which makes it inaccessible to some dependents. For example, [eslint-plugin-compat](https://www.npmjs.com/package/eslint-plugin-compat) which accounts for over 82% of `mdn-browser-compat-data` installs (175K installs/week out of 213K/week) [has to support NodeJS 9](https://github.com/amilajack/eslint-plugin-compat/blob/9e35ab8c1b796dec9e994d01c27a37f94799d650/package.json#L89).

"Bundled" releases would be compatible with any NodeJS and browser versions (because it's just a JSON) and would simultaneously allow BCD to drop any NodeJS versions as soon as needed.

**Comment by vinyldarkscratch** [comment]
+1 for this -- the reduced size and load times are incentive enough!  Just thinking, we'll probably need to bump at least the minor version, if not the major version.

**Comment by bershanskiy** [comment]
I'm closing this since we recently landed #7398 and it was already released on npm.
